app/components/main/main.component.ts

```diff
@@ -217,9 +224,11 @@ export class MainComponent implements OnInit, MenuEventListener, REPLDelegate, T
                                 throw "invalid file format version";
                             }
 
-                            this.newFile(filename.substring(Math.max(filename.lastIndexOf("/"),
-                                filename.lastIndexOf("\\")) + 1),
-                                new Core.Graph(this.pluginService.getPlugin("dfa.sinap.graph-kind")));
+                            this.pluginService.getPlugin("dfa.sinap.graph-kind").then((plugin) => {
```

Comment:
This should **not** be in a constants file. These are strings that will eventually be loaded based on the plugin.